[Docs] Add how to guide for customising VizroAI (#438)
mckinsey/vizro: vizro-ai/docs/pages/user-guides/customize-vizro-ai.md

> # How to customize VizroAI

Reviewer: @nadijagraca I'm not sure if we need a new page for this content. Some of it is already in the FAQ https://vizro.readthedocs.io/projects/vizro-ai/en/latest/pages/explanation/faq/ and I wonder if I should just take what you've added and move it into the relevant Q&A pair (if it's not already stated). WDYT?

Reviewer 2: @stichbury I made a slightly different suggestion below, let me know what you think. That would kill the FAQs alltogther

Reviewer: I'll take a look. @nadijagraca I can work on your branch if that's OK and move the content around as needed.

> # How to customize VizroAI

Reviewer 2: ```suggestion
# How to customize your model
```

Reviewer 2: I think this might actaully be the better title - based on that I would re-read below, and change some if the things accordingly

This guide explains how to customize `VizroAI`.

> `VizroAI` accepts a single argument named `model`. This `model` argument can either be a string that specifies the name of the `ChatOpenAI` model or an instantiated `ChatOpenAI` model.

Reviewer 2: ```suggestion
`VizroAI` accepts a single argument named `model`. This `model` argument can either be a `str` that specifies the name of the `ChatOpenAI` model or an instantiated `ChatOpenAI` model.
```

We'll delve into customizing the `ChatOpenAI` model later; for now, let's concentrate on the default usage of `VizroAI`.

## Default usage

> `VizroAI`can be initialized without any arguments, in which case it will, by default, use OpenAI model `"gpt-3.5-turbo"` with a temperature setting of 0.

Reviewer 2: ```suggestion
`VizroAI` can be initialized without any arguments, in which case it will, by default, use the OpenAI model `"gpt-3.5-turbo"` with a temperature setting of 0.
```

`"gpt-3.5-turbo"` points to the latest model offering enhanced speed and accuracy in generating responses in requested formats, while maintaining cost-effective performance.

> ## Choosing OpenAI model

Reviewer 2: ```suggestion
## Customize by `str`
```

> When choosing OpenAI model, there are few things to keep in mind; the complexity of the task, cost associated with it and the desired response time.

Reviewer 2: ```suggestion
When choosing an OpenAI model, there are few things to keep in mind; the complexity of the task, cost associated with it and the desired response time.
```

> When choosing OpenAI model, there are few things to keep in mind; the complexity of the task, cost associated with it and the desired response time.

Reviewer 2: I would add the FAQ section before here, with most of the content written there. PLease re-read to see if everything fits. Then the read know which string to but.

Then I really like you explanation about the different series, so the reader finds out more about what he might choose.

Finally, I would delete the FAQ sections alltogether - @stichbury might have a few on where to put the authors

**gpt-3.5** model series are good in both general tasks and chat-specific applications. These models have lower price point and higher speeds for providing answers.

However, for more demanding tasks, consider upgrading to the **gpt-4** models. While they are part of a more capable GPT model series, their response time is slower than gpt-3.5 models, and they come at a higher cost.

> In the example below, we customized `VizroAI` by providing OpenAI model name in a string form. ( see for [models currently supported by Vizro-AI](../explanation/faq.md#which-llms-are-supported-by-vizro-ai)).

Reviewer 2: ```suggestion
The below example shows how to customize your model in `str` form.
```

!!! example "Customize with string"

    === "python"
        ```py linenums="1"
        from vizro_ai import VizroAI

        vizro_ai = VizroAI(model="gpt-3.5-turbo-0125")
        ```

> ## Customize ChatOpenAI instance

Reviewer 2: ```suggestion
## Customize by providing a `ChatOpenAI` instance
```

> ## Customize ChatOpenAI instance

Reviewer 2: I would add somewhere in this section on where to import from, and what we support again, with a hint that we are working on more vendors.

> The `model` parameter in `VizroAI` can take an initialized `ChatOpenAI` model, allowing for flexible customization by users. In the example below, we'll be customizing the `ChatOpenAI` instance.

Reviewer 2: ```suggestion
The `model` parameter in `VizroAI` can also take an initialized `ChatOpenAI` model, allowing for flexible customization by users. In the example below, we'll be customizing the `ChatOpenAI` instance.
```

The `"gpt-3.5-turbo-0125"` model from OpenAI as `model_name` for `ChatOpenAI`, which offers improved response accuracy. To ensure a deterministic answer to our queries, we've set the temperature to 0.

> If you prefer more creative responses, you can raise the temperature to a maximum of 1.

Reviewer 2: ```suggestion
If you prefer more creative (but potentially more unstable) responses, you can raise the temperature to a maximum of 1.
```

> # How to customize Vizro-AI

Reviewer 2: It really is a guide on how to customize the model, so I would probably clearlt label it like that

Reviewer 2: (see also now the chosen title in the TOC)

Reviewer: I went with "How to customize model usage"

> ## Which LLMs are supported by Vizro-AI?

Reviewer 2: ```suggestion
## Supported models
```

> Vizro-AI currently supports OpenAI models as follows:

Reviewer 2: can we also add an overview link to the OpenAI models

* Consider upgrading to the **gpt-4** models for more demanding tasks. While they are part of a more capable GPT model series, their response time is slower than gpt-3.5 models, and they come at a higher cost.

Refer to the [OpenAI documentation for more about model capabilities](https://platform.openai.com/docs/models/overview) and [pricing](https://openai.com/pricing).

## Default initialization
`VizroAI`can be initialized without any arguments, in which case it uses `"gpt-3.5-turbo"` by default, with a temperature setting of 0. `"gpt-3.5-turbo"` offers enhanced speed and accuracy, and generates responses in requested formats while maintaining cost-effective performance.

## Customization at initialization
To customize the model, you can pass `VizroAI` a single argument named `model`, which can either be a string that specifies the name of a `ChatOpenAI` model or an instantiated `ChatOpenAI` model.

The example below uses the OpenAI model name in a string form:

!!! example "Customize with string"

    === "python"
        ```py linenums="1"
        from vizro_ai import VizroAI

        vizro_ai = VizroAI(model="gpt-3.5-turbo-0125")
        ```

> The example below customizes the `ChatOpenAI` instance, passing the `"gpt-3.5-turbo-0125"` model from OpenAI as `model_name` for `ChatOpenAI`, which offers improved response accuracy.

Reviewer 2: ```suggestion
The example below customizes the `ChatOpenAI` instance further beyond the chosen default from the string instatiation. We pass the `"gpt-3.5-turbo-0125"` model from OpenAI as `model_name` for `ChatOpenAI`, which offers improved response accuracy.
```

<!-- vale off -->
To ensure a deterministic answer to our queries, we've set the temperature to 0. If you prefer more creative (but potentially more unstable) responses, you can raise the temperature to a maximum of 1.
<!-- vale on -->

> !!! example "model customization"

Reviewer 2: ```suggestion
!!! example "Customize with instantiated model"
```
